src/bidiMapper/domains/protocol/bidiProtocolTypes.ts

```diff
@@ -402,6 +408,86 @@ export namespace Script {
     text: string;
   };
 
+  export type RealmInfo =
+    | WindowRealmInfo
+    | DedicatedWorkerRealmInfo
+    | SharedWorkerRealmInfo
+    | ServiceWorkerRealmInfo
+    | WorkerRealmInfo
+    | PaintWorkletRealmInfo
+    | AudioWorkletRealmInfo
+    | WorkletRealmInfo;
+
+  export type BaseRealmInfo = {
+    realm: Realm;
+    origin: string;
+  };
+
+  export type WindowRealmInfo = BaseRealmInfo & {
+    type: 'window';
+    context: CommonDataTypes.BrowsingContext;
+    sandbox?: string;
+  };
+
+  export type DedicatedWorkerRealmInfo = BaseRealmInfo & {
+    type: 'dedicated-worker';
+  };
+
+  export type SharedWorkerRealmInfo = BaseRealmInfo & {
+    type: 'shared-worker';
+  };
+
+  export type ServiceWorkerRealmInfo = BaseRealmInfo & {
+    type: 'service-worker';
+  };
+
+  export type WorkerRealmInfo = BaseRealmInfo & {
+    type: 'worker';
+  };
+
+  export type PaintWorkletRealmInfo = BaseRealmInfo & {
+    type: 'paint-worklet';
+  };
+
+  export type AudioWorkletRealmInfo = BaseRealmInfo & {
+    type: 'audio-worklet';
+  };
+
+  export type WorkletRealmInfo = BaseRealmInfo & {
+    type: 'worklet';
+  };
+
+  const RealmTypeSchema = zod.enum([
+    'window',
+    'dedicated-worker',
+    'shared-worker',
+    'service-worker',
+    'worker',
```

Comment:
if the order does not matter, can we sort this list?

Reply:
The order is aligned with [the order in spec](https://w3c.github.io/webdriver-bidi/#type-script-RealmInfo).